src/node/net.ts

```diff
@@ -1110,10 +1110,21 @@ async function startRead(socket: SocketClass): Promise<void> {
   const reader = socket._handle.reader;
   try {
     while (socket._handle.reading === true) {
+      const generatedBuffer = socket[kBufferGen]?.();
+
+      if (
```

Comment:
Likely worthwhile mentioning the conditions under which kBufferGen will return a zero-length buffer.